ts/features/itwallet/common/utils/__tests__/itwClaimsUtils.test.ts

```diff
@@ -468,3 +469,135 @@ describe("getCredentialStatus", () => {
+    const result = DateWithoutTimezoneClaim.decode(input);
+
+    expect(E.isRight(result)).toBe(true);
+    if (E.isRight(result)) {
+      const date = result.right;
+      expect(date.toISOString()).toBe("2024-12-31T00:00:00.000Z");
+    }
+  });
+
+  test("should validate a leap year date (February 29)", () => {
+    const input = "2024-02-29"; // Anno bisestile
+    const result = DateWithoutTimezoneClaim.decode(input);
+
+    expect(E.isRight(result)).toBe(true);
+    if (E.isRight(result)) {
+      const date = result.right;
+      expect(date.toISOString()).toBe("2024-02-29T00:00:00.000Z");
+    }
+  });
+
+  test("should validate a string representing January 1st", () => {
+    const input = "2024-01-01"; // Primo giorno dell'anno
+    const result = DateWithoutTimezoneClaim.decode(input);
+
+    expect(E.isRight(result)).toBe(true);
+    if (E.isRight(result)) {
+      const date = result.right;
+      expect(date.toISOString()).toBe("2024-01-01T00:00:00.000Z");
+    }
+  });
+
+  test("should fail for empty string input", () => {
+    const input = ""; // Stringa vuota
+    const result = DateWithoutTimezoneClaim.decode(input);
+
+    expect(E.isRight(result)).toBe(false);
+    if (E.isRight(result) === false) {
+      expect(result.left[0].message).toBe("Date is not in the correct format");
+    }
+  });
+
+  test("should fail for a string with invalid characters", () => {
+    const input = "2024-11-abc"; // Input con caratteri non validi
+    const result = DateWithoutTimezoneClaim.decode(input);
+
+    expect(E.isRight(result)).toBe(false);
+    if (E.isRight(result) === false) {
+      expect(result.left[0].message).toBe("Date is not in the correct format");
+    }
+  });
+
+  test("should fail if the year is less than four digits", () => {
+    const input = "24-11-19"; // Anno non valido
+    const result = DateWithoutTimezoneClaim.decode(input);
+
+    expect(E.isRight(result)).toBe(false);
+    if (E.isRight(result) === false) {
+      expect(result.left[0].message).toBe("Date is not in the correct format");
+    }
+  });
+
+  test("should fail if the year exceeds reasonable bounds", () => {
+    const input = "10000-01-01"; // Anno troppo grande
+    const result = DateWithoutTimezoneClaim.decode(input);
+
+    expect(E.isRight(result)).toBe(false);
+    if (E.isRight(result) === false) {
+      expect(result.left[0].message).toBe("Date is not in the correct format");
+    }
+  });
+
+  test("should serialize and deserialize a Date instance correctly", () => {
+    const input = new Date(Date.UTC(2024, 10, 19)); // 19 Novembre 2024
+    const encoded = DateWithoutTimezoneClaim.encode(input);
+    expect(encoded).toBe("2024-11-19");
+
+    const decoded = DateWithoutTimezoneClaim.decode(encoded);
+    expect(E.isRight(decoded)).toBe(true);
+    if (E.isRight(decoded)) {
+      const date = decoded.right;
+      expect(date.toISOString()).toBe("2024-11-19T00:00:00.000Z");
+    }
+  });
+});
```

Comment:
Let's group these test cases into one, so it should be easier to test with different timezones.
For example:

```
it("should handle dates without timezone" () => { .. })
```

with all the tests inside